### ISSUE TITLE
Add multi-gpu training example script

### ISSUE THREAD

**Issue by qubvel** [qubvel-org/segmentation_models.pytorch#913]
It would be great to have more examples for fine-tuning in the library!

Similar to current examples for [binary-segmentaion](https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/binary_segmentation_intro.ipynb) and [multi-label](https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/cars%20segmentation%20(camvid).ipynb) segmentation would be great to have Multi-GPU training example. This can be handled with [PyTorch-Lightning](https://lightning.ai/docs/pytorch/stable/accelerators/gpu_intermediate.html) or with pure Pytorch.

It might be a `.py` script or Jupyter notebook. The example should showcase how to

 - fine-tune a model with pytorch-lightning (or any other fine-tuning framework, even a plain pytorch)
 - compute metrics with correct gathering of results (maybe [torchmetrics](https://lightning.ai/docs/torchmetrics) can be utilized)
 - visualize results

In case anyone wants to work on this you are welcome! Just notify in this issue, and then ping me to review a PR when it's ready.
